### ISSUE TITLE
is Union a _SpecialForm?

### ISSUE THREAD

**Issue by comex** [microsoft/pyright#10091]
This code typechecks under mypy and fails under pyright:
```python
from typing import Union, _SpecialForm
x: _SpecialForm = Union[int, str]
```
I don't know whether this code should be considered correct.  After all, `isinstance(Union[int, str], _SpecialForm)` returns False.  However, the `typedload` package seems to rely on this for its own type annotations.  Given this code:
```python
from typing import Union, Any
import typedload
x: Any = typedload.load(42, Union[int, str])
```
if I run
```
uv run --with typedload --with mypy mypy /tmp/test.py
```
it succeeds, but if I run
```
uv run --with typedload --with pyright pyright /tmp/test.py
```
I get
```
  /tmp/test.py:3:29 - error: Argument of type "UnionType" cannot be assigned to parameter "type_" of type "type[T@load] | _SpecialForm" in function "load"
    Type "UnionType" is not assignable to type "type[T@load] | _SpecialForm"
      Type "UnionType" is not assignable to type "type[T@load]"
      "UnionType" is not assignable to "_SpecialForm" (reportArgumentType)
1 error, 0 warnings, 0 informations 
```
Reproduced with pyright 1.1.396, mpy 1.15.0, typedload-2.37.

**Comment by erictraut** [comment]
Pyright is working correctly here, so I don't consider this a bug.

The symbol `typing.Union` is a special form, but the expression `Union[int, str]` is not assignable to the type `_SpecialForm`.

The `typing._SpecialForm` symbol is an undocumented internal implementation detail of the type system. It should not be imported or used by any modules outside of the stdlib typeshed stubs. Not surprisingly, there is no mention of `_SpecialForm` in the [typing spec](https://typing.python.org/en/latest/spec/index.html).

I suspect that the maintainers of this library intended to define a type alias, in which case they should use one of the mechanisms specified in the typing spec to define [type aliases](https://typing.python.org/en/latest/spec/aliases.html). For example, `x: TypeAlias = Union[int, str]` or `type x = int | str`.

If this library is important to you, I recommend filing a bug to let its maintainers know about this issue. Hopefully they can get it fixed quickly for you.
